pkg/build/generator/generator.go

```diff
@@ -5,19 +5,25 @@ import (
 	"regexp"
 	"strings"
 
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
 	"github.com/golang/glog"
 
 	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
 	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
 	imageapi "github.com/openshift/origin/pkg/image/api"
 )
 
 // BuildGenerator is a central place responsible for generating new Build objects
 // from BuildConfigs and other Builds.
 type BuildGenerator struct {
-	Client GeneratorClient
+	Client          GeneratorClient
+	ServiceAccounts kclient.ServiceAccountsNamespacer
+	Secrets         kclient.SecretsNamespacer
+	BuilderAccount  *kapi.ServiceAccount
```

Comment:
@ncdc just helps the testing above.... I try to fix the tests tomorrow and make it not exported.